library/data/registry-entry-attributes/src/main/java/org/quiltmc/qsl/registry/attribute/api/DispatchedType.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2021 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.registry.attribute.api;
+
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+import java.util.function.Function;
+
+/**
+ * Utility interface used for {@link RegistryEntryAttribute#createDispatched(Registry, Identifier, Function, DispatchedType)}.<p>
+ *
+ * This allows for polymorphic attribute types!
+ */
+public interface DispatchedType {
```

Comment:
I feel the purpose of this would be better explained using part of the testmod you have made for this.

Reply:
I've tried to improve this interface's Javadoc using the testmod as an example.